Arduino-Bling/RobotBling/RobotBling.ino

```diff
@@ -0,0 +1,209 @@
+ * Reminder: we don't have 'w' component in our neopixels -- rgb components are represented
+ */
+struct rgbColor
+{
+  // initializers
+  rgbColor() { color = 0x000000; } // if 'w' component was present: 0x00000000
+  rgbColor(uint32_t c) : color(c) {}
+  rgbColor(uint8_t red, uint8_t green, uint8_t blue) : b(blue), g(green), r(red) {}
+
+  union {
+    uint32_t color; // full color
+    struct
+    {
+      uint8_t b; // blue component
+      uint8_t g; // green component
+      uint8_t r; // red component
+    };
+  };
+};
+
+// Colors used in the animations
+const rgbColor rgbColor_OFF(0, 0, 0);
+const rgbColor rgbColor_RED(255, 0, 0);
+const rgbColor rgbColor_GREEN(0x00ff00);
+const rgbColor rgbColor_BLUE(0, 0, 255);
+const rgbColor rgbColor_YELLOW(0xffff00);
+const rgbColor rgbColor_ORANGE(255, 64, 0);
+const rgbColor rgbColor_WHITE(0xffffff);
+
+const rgbColor rgbColor_MAGENTA(0xff00ff);
+const rgbColor rgbColor_MAGENTA_DIM(0x330033);
+
+/*
+ * Animation library
+ *  - List of animation used by Spartronics -- MUST match LED subsystem BlingStates
+ *  - loop() matches the animation to the command entered on the serial port
+ */
+enum
+{
+  OFF = 0,
+  DISABLED,
+  SOME_BLING_STATE_2,
+  SOME_BLING_STATE_3,
+
+  ANIMATION_COUNT // The number of animations
+};
+
+/*
+ * Magic delay function for driving animation library
+ * - Note that the delay() function mapped to _delay()
+ * - Anytime there is a delay() introduced to animations, it allows to check for new commands
+ * - commands drive the animation demo -- if a new command detected, we jump to the beginning
+ * of the loop -- see loop()
+ *
+ * important: _delay() is listening for printable characters '0' and
+ * above. This matches the message sent by robot's LED subsystem
+ */
+
+// A global variable to hold the current command to be executed
+uint8_t currentCommand = OFF;
+
+// The breadcrumb to get us back to the beginning of the loop
+jmp_buf env;
+
+// Our magic delay function
+void _delay(uint16_t timeout)
+{
+  while (timeout != 0)
+  {
+    delay(1);
+    timeout--;
+
+    // If there are any characters available on the serial port, read them
+    while (Serial.available() > 0)
+    {
+      char commandCharacter;
+      commandCharacter = Serial.read();
+      // use the ASCII table to identify the >= commandCharacter order
+      if (isprint(commandCharacter) && (commandCharacter >= '0'))
+      {
+        uint8_t command = commandCharacter - '0';
+        if (command < ANIMATION_COUNT)
+        {
+          // Found a valid command!
+          if (command != currentCommand)
+          {
+            // It's a new command!
+            currentCommand = command;
+            // Jump back to the top of the loop
+            longjmp(env, 1);
+          }
+        }
+      }
+    }
+  }
+}
+
+// redefining delay function
+#define delay _delay
+
+/*
+ * Long waited setup() -- clear the strip and set brightness level
```

Comment:
What does 'Long waited' mean?
Should be 'Initialize system state, set up serial communications, and clear the LED strip'